parquet-column/src/main/java/org/apache/parquet/column/values/factory/ValuesWriterFactoryParams.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.factory;
+
+import org.apache.parquet.bytes.ByteBufferAllocator;
+import org.apache.parquet.column.ParquetProperties.WriterVersion;
+
+/**
+ * Encapsulates parameters needed to create new ValuesWriter classes as part of
+ * ValuesWriterFactory
+ */
+public class ValuesWriterFactoryParams {
+
+  public ValuesWriterFactoryParams(
+    WriterVersion writerVersion,
```

Comment:
I guess what I'm getting at is that these parameter values are only used specifically within DefaultValuesWriterFactory and then passed into that implementation's writers (from what I can tell in the PR, it's possible I'm totally wrong). While it's likely that most other implementations would need similar/the same values in the end, it's totally possible that they could want to get them from elsewhere.

You could always just ignore the default ValueWritersFactoryParams if you're using a totally different method (for example let's say your writer infers page size somehow) to come up with your values but that just seems kind of suboptimal to just having each writerfactory provide their own implementation of FactoryParams.

I'm happy to clarify if that doesn't make sense or hear if you disagree.